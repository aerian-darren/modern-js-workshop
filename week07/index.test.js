import {
    appendToTitle,
    createKeyboard,
    makeButton,
    makeButtons,
} from "./index.js";
import emojis from "../lib/emoji.js";

it("appends to the title", () => {
    document.body.innerHTML = "<h1 id='heading'></h1>";
    appendToTitle("Hello");
    expect(document.querySelector("h1").textContent).toEqual("Hello");
    appendToTitle("World");
    expect(document.querySelector("h1").textContent).toEqual("HelloWorld");
});

it("creates buttons in the form", () => {
    document.body.innerHTML =
        "<h1 id='heading'></h1><form id='keyboard'></form>";
    createKeyboard(emojis);
    expect(document.querySelector("form button")).toBeTruthy();
});

it("creates a button", () => {
    const button = makeButton("ðŸ˜»");

    expect(button).toMatchSnapshot();
});

it("creates lots of buttons", () => {
    const buttons = makeButtons(emojis);

    expect(buttons.length).toEqual(emojis.length);
});
