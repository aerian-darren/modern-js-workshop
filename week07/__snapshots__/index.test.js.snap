// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a button 1`] = `
<button>
  ðŸ˜»
</button>
`;
